src/server/data/sykmeldt/postSlettArbeidsoppgaveSM.ts

```diff
@@ -0,0 +1,50 @@
+import { isMockBackend } from "environments/publicEnv";
+import { NextApiResponse } from "next";
+import activeMockSM from "server/data/mock/activeMockSM";
+import { deleteOppgave } from "server/service/oppfolgingsplanService";
+import { handleQueryParamError } from "server/utils/errors";
+import serverLogger from "server/utils/serverLogger";
+import { getOppfolgingsplanTokenX } from "server/utils/tokenX";
+import { IAuthenticatedRequest } from "../../api/IAuthenticatedRequest";
+
+export const postSlettArbeidsoppgaveSM = async (
+  req: IAuthenticatedRequest,
+  res: NextApiResponse,
+  next: () => void
+) => {
+  const { oppfolgingsplanId, arbeidsoppgaveId } = req.query;
+
+  if (typeof oppfolgingsplanId !== "string") {
+    return handleQueryParamError(oppfolgingsplanId);
+  }
+
+  if (typeof arbeidsoppgaveId !== "string") {
+    return handleQueryParamError(arbeidsoppgaveId);
+  }
+
+  if (isMockBackend) {
+    const { oppfolgingsplanId, arbeidsoppgaveId } = req.query;
+
+    const aktivPlan = activeMockSM.oppfolgingsplaner.find(
+      (plan) => plan.id == Number(oppfolgingsplanId)
+    );
+    const aktivPlanIndex = activeMockSM.oppfolgingsplaner.indexOf(aktivPlan!!);
+    const filteredArbeidsoppgaveListe =
+      aktivPlan!!.arbeidsoppgaveListe!!.filter(
+        (arbeidsoppgave) =>
+          arbeidsoppgave.arbeidsoppgaveId != Number(arbeidsoppgaveId)
+      );
+
+    activeMockSM.oppfolgingsplaner[aktivPlanIndex].arbeidsoppgaveListe =
+      filteredArbeidsoppgaveListe;
+  } else {
+    const oppfolgingsplanTokenX = await getOppfolgingsplanTokenX(req);
```

Comment:
Burde den ikke sende inn `idportentoken` som argument? Hvis ikke burde vi endre `getOppfolgingsplanTokenX` til å gjøre det.

Reply:
- [ ] Addresser

Reply:
Du tenker at man sender inn `idportentoken` som et argument i stedet for `req: IAuthenticatedRequest`?

Reply:
Hva betyr denne?

> * [ ]  Addresser